spec/selenium/conferences/big_blue_button_conference_spec.rb

```diff
@@ -0,0 +1,104 @@
+# the terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, version 3 of the License.
+#
+# Canvas is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <http://www.gnu.org/licenses/>.
+
+require_relative '../common'
+require_relative '../helpers/conferences_common'
+require_relative '../helpers/public_courses_context'
+
+describe 'BigBlueButton conferences' do
+  include_context 'in-process server selenium tests'
+  include ConferencesCommon
+  include WebMock::API
+
+  bbb_endpoint = 'bbb.instructure.com'
+  bbb_secret = '8cd8ef52e8e101574e400365b55e11a6'
+  bbb_fixtures = {
+    :get_recordings => {
+      'meetingID' => 'instructure_web_conference_3Fn2k10wu0jK7diwJHs2FkDU0oXyX1ErUZCavikc',
+      'checksum' => 'ffa33deaa16e039bcbd84df90ed39196e518c69e'
+    },
+    :delete_recordings => {
+      'recordID' => '0225ccf234655ae60658ccac1e272d48781b491c-1511812307014',
+      'checksum' => 'f9a2b4c9b3e0dd611c1b541eee8bb32309a327ae'
+    }
+  }
+
+  before(:once) do
+    initialize_big_blue_button_conference_plugin bbb_endpoint, bbb_secret
+    course_with_teacher(name: 'Teacher Bob', active_all: true)
+    course_with_ta(name: 'TA Alice', course: @course, active_all: true)
+    4.times do |i|
+      course_with_student(name: "Student_#{i + 1}", course: @course, active_all: true)
+    end
+  end
+
+  before(:each) do
+    user_session(@teacher)
+    get conferences_index_page
+  end
+
+  after { close_extra_windows }
+  context 'when a conference is open' do
+    before(:once) {
+      stub_request(:get, 'bbb.instructure.com/bigbluebutton/api/create')
+    }
+
+    context 'and the conference has no recordings' do
+      before(:once) {
+        stub_request(:get, 'bbb.instructure.com/bigbluebutton/api/getRecordings').
+        with(query: bbb_fixtures[:get_recordings]).
+        to_return(:body => big_blue_button_mock_response('get_recordings', 'none'))
+        @conference = create_big_blue_button_conference(bbb_fixtures[:get_recordings]['meetingID'])
+      }
+
+      it 'should not include list with recordings', priority: '2' do
+        verify_conference_does_not_include_recordings
+      end
+    end
+
+    context 'and the conference has recordings' do
+      before(:once) {
+        stub_request(:get, 'bbb.instructure.com/bigbluebutton/api/getRecordings').
+        with(query: bbb_fixtures[:get_recordings]).
+        to_return(:body => big_blue_button_mock_response('get_recordings', 'two'))
+        @conference = create_big_blue_button_conference(bbb_fixtures[:get_recordings]['meetingID'])
+      }
+
+      it 'should include list with recordings', priority: '2' do
```

Comment:
This test is failing.

```
BigBlueButton conferences when a conference is open and the conference has recordings should include list with recordings
  Failure/Error: expect(first_conference_in_list(new_conference_list)).to include_text 'Recording'
  
    expected #<Selenium::WebDriver::Element:0x..fc605d7237afed1aa id="{fe2a1da1-038f-40ae-b5a5-bcfbfcf5f381}"> text to include Recording, actual text was: BigBlueButton Conference
    Start
    Settings
```

Reply:
This one, I don't have idea why is happening. The text shown here:
```
expected #<Selenium::WebDriver::Element:0x..fc605d7237afed1aa id="{fe2a1da1-038f-40ae-b5a5-bcfbfcf5f381}"> text to include Recording, actual text was: BigBlueButton Conference
    Start
    Settings
```
If we were executing the opposite test (verify_conference_does_not_include_recordings) it should be like this:
```
 expected #<Selenium::WebDriver::Element:0x..fc484bc3484ccba84 id="4"> text to NOT include Recording, actual text was: BigBlueButton Conference
       2 Recordings
       Start
       Settings
       Conferencia de Development 101
       Links to an external site.
        |  Duration: 1 minute
```
Which makes me think that it is either (as for the previus one) that the assets used are not the new ones, or that webmock is not stubbing the request.

But here I don't know what to do as the local tests are passing
```
RAILS_ENV=test rake spec:selenium
```

Reply:
Hi @jfederico I'll retrigger another pass of these tests to make sure we get the same failures. I can also try running these tests locally to see if I get any different results.

For reference, here's a screenshot of the page when the test is failing:
![screen shot 2018-02-12 at 3 52 55 pm](https://user-images.githubusercontent.com/6925979/36121771-cffcb39c-100c-11e8-99cb-cdaee1fcbad5.png)

Reply:
OK, just ran this spec file locally after compiling assets, and my results agree with the build. I am seeing these two specs fail:

```
Failed examples:

rspec ./spec/selenium/conferences/big_blue_button_conference_spec.rb:97 # BigBlueButton conferences when a conference is open and the conference has one recording and it is deleted should remove recording from the list
rspec ./spec/selenium/conferences/big_blue_button_conference_spec.rb:81 # BigBlueButton conferences when a conference is open and the conference has recordings should include list with recordings

Randomized with seed 15803
```
Can you try running the specs with the seed set to 15803 (to match my run above)?

`rspec spec/selenium/conferences/big_blue_button_conference_spec.rb --seed 15803`

Reply:
It turned out no to be an issue with the fixture files themselves but with the way the request was done. The model makes requests to BBB as https and it is hardcoded. The specs were considering http as it was not specified.

It should be good now. Still missing the test to coffescript.

Reply:
Alright, great news! I took your newest commit and pushed it through tests, and they all passed. Once we have the Coffeescript tests for ConferenceView, I'll get this back in front of reviewers.

Thanks!